crates/biome_css_analyze/src/lint/nursery/use_sorted_properties.rs

```diff
@@ -0,0 +1,409 @@
+        language: "css",
+        recommended: true,
+        fix_kind: FixKind::Safe,
+    }
+}
+
+pub struct UseSortedPropertiesState {
+    block: CssDeclarationOrRuleBlock,
+    can_be_sorted: bool,
+    first_out_of_order_pair: Option<(NodeWithPosition, NodeWithPosition)>,
+}
+
+impl Rule for UseSortedProperties {
+    type Query = Ast<CssDeclarationOrRuleBlock>;
+    type State = UseSortedPropertiesState;
+    type Signals = Option<Self::State>;
+    type Options = ();
+
+    fn run(ctx: &RuleContext<Self>) -> Option<Self::State> {
+        let query_result = ctx.query();
+        let items = to_block_items_with_position(query_result);
+
+        let first_out_of_order_pair = get_first_out_of_order_pair(&items);
+        first_out_of_order_pair.as_ref()?;
+
+        Some(UseSortedPropertiesState {
+            block: query_result.clone(),
+            can_be_sorted: !contains_shorthand_after_longhand(&items)
+                && !contains_unknown_property(&items),
+            first_out_of_order_pair,
+        })
+    }
+
+    fn diagnostic(_: &RuleContext<Self>, state: &Self::State) -> Option<RuleDiagnostic> {
+        if !state.can_be_sorted || state.first_out_of_order_pair.is_none() {
+            return None;
+        }
+
+        if let Some((a, b)) = &state.first_out_of_order_pair {
+            let a_description = node_short_description(a);
+            let b_description = node_short_description(b);
+            return Some(RuleDiagnostic::new(
+                rule_category!(),
+                state.block.range(),
+                markup! {
+                    "Properties should be sorted: "<Emphasis>{ a_description }</Emphasis>" should be before "<Emphasis>{ b_description }</Emphasis>"."
+                },
+            ));
+        }
+
+        None
+    }
+
+    fn action(ctx: &RuleContext<Self>, state: &Self::State) -> Option<CssRuleAction> {
+        if !state.can_be_sorted || state.first_out_of_order_pair.is_none() {
+            return None;
+        }
+
+        let mut mutation = ctx.root().begin();
+
+        let original_items = to_block_items_with_position(&state.block);
+        let mut sorted_items = original_items.clone();
+        sorted_items.sort_by(|a, b| sort_info(&a.node).cmp(&sort_info(&b.node)));
+
+        let pairs = original_items.iter().zip(sorted_items.iter());
+
+        for (item_to_replace, replacement) in pairs {
+            if item_to_replace.position != replacement.position {
+                mutation.replace_node_discard_trivia(
+                    item_to_replace.node.clone(),
+                    replacement.node.clone(),
+                );
+            }
+        }
+
+        return Some(RuleAction::<CssLanguage>::new(
+            ActionCategory::QuickFix,
+            ctx.metadata().applicability(),
+            markup! { "Sort these properties" }.to_owned(),
+            mutation,
+        ));
+    }
+}
+
+#[derive(Clone)]
+struct NodeWithPosition {
+    position: usize,
```

Comment:
What does this position represent here? Without documentation, it's hard to understand. I checked where it's used, and it seems it's used in the diagnostics. Can we clarify? 

On a side note, I don't think telling users about the concept of "position" is correct. We shouldn't have it. Instead, let's save the `TextRange` of the node, so in the diagnostic we can point the user to the exact node.